app/helpers/sources_helper.rb

```diff
@@ -13,4 +13,35 @@ def render_media_asset
       render partial: 'video'
     end
   end
+
+  ##
+  # Render a back link to the referring page if the referring page is a guide or
+  # set.  Links from guides or sets to sources may be created by admins, so the
+  # applicaiton cannot expect that back URI's can be included in params.
+  # Therefore, it is necessary to parse the referring URI to see if it is a
+  # guide or set, and render a back link accordingly.
+  #
+  # @return [String] HTML back link
+  def render_back_link
+    return unless request.referer.present?
+    uri = URI(request.referer)
+    # return if referrer is an offsite link
+    return unless uri.host == request.host
```

Comment:
Is this going to return with no action the way we want it to if there's a link from somewhere else on the same domain, e.g. the main DPLA site linking to a source page?